### PR TITLE
added User and Role

### PR DESCRIPTION
**Pull request by Ilya-brinchuk**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -26,11 +30,20 @@
...
         <dependency>
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-jdbc</artifactId>
+            <version>2.0.7.RELEASE</version>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: do we need this dependency?

File: src/main/java/bankservice/demo/controller/DataInitializer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.time.LocalDate;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataInitializer {
+    private RoleService roleService;
+    private UserService userService;
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/controller/DataInitializer.java]:
```suggestion
    private final RoleService roleService;
    private final UserService userService;
```

File: src/main/java/bankservice/demo/exception/DataProcessingException.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package bankservice.demo.exception;
+
+public class DataProcessingException extends RuntimeException {
+
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/exception/DataProcessingException.java]: redundant empty line

File: src/main/java/bankservice/demo/repository/RoleRepository.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import bankservice.demo.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("select r from roles r where r.roleName = :name")
+    Optional<Role> getRoleByRoleName(@Param("name") String name);
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/repository/RoleRepository.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/java/bankservice/demo/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("select u from users u inner join fetch u.roles where u.id = :id")
+    Optional<User> findByIdFetchRoles(@Param(value = "id") Long id);
+
+    @Query("select u from users u inner join fetch u.roles where u.phoneNumber = :number")
+    Optional<User> findUserByPhoneNumber(@Param(value = "number") String number);
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/repository/UserRepository.java]:
```suggestion
    Optional<User> findById(Long id);

    @Query("select u from users u inner join fetch u.roles where u.phoneNumber = :number")
    Optional<User> findByPhoneNumber(String number);
```

we can simplify this a bit

File: src/main/java/bankservice/demo/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package bankservice.demo.service;
+
+import bankservice.demo.model.User;
+
+public interface UserService {
+    User create(User user);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/service/UserService.java]: see checklist

File: src/main/java/bankservice/demo/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+    User get(Long id);
+
+    void delete(Long id);
+
+    User findUserByPhoneNumber(String phoneNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/service/UserService.java]:
```suggestion
    User getByPhoneNumber(String phoneNumber);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/bankservice/demo/repository/UserRepository.java [hunk 8]
```diff
@@ -4,12 +4,10 @@
 import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.data.repository.query.Param;
 
 public interface UserRepository extends JpaRepository<User, Long> {
-    @Query("select u from users u inner join fetch u.roles where u.id = :id")
-    Optional<User> findByIdFetchRoles(@Param(value = "id") Long id);
+    Optional<User> findById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/repository/UserRepository.java:10]: you need to join fetch roles here as well - that was right
I meant that we can remove `@Param` annotation

File: src/main/java/bankservice/demo/service/RoleService.java [hunk 9]
```diff
@@ -5,5 +5,5 @@
 public interface RoleService {
     Role save(Role role);
 
-    Role getByName(String name);
+    Role getByName(Role.RoleName roleName);
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/service/RoleService.java]: you should not change this method
on the service level we get String - and RoleName on the dao layer

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/bankservice/demo/controller/DataInitializer.java [hunk 10]
```diff
@@ -30,6 +30,7 @@ public void inject() {
         roleAdmin.setRoleName(Role.RoleName.ADMIN);
         roleService.save(roleUser);
         roleService.save(roleAdmin);
+        System.out.println(roleService.getByName("ADMIN"));
```
Comment (KseniiaMakarova) [inline src/main/java/bankservice/demo/controller/DataInitializer.java:33]: great, we may remove this)

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**